### ISSUE TITLE
`non-augmented-assignment` (`PLR6104`) - false positive on operators where the variable is on the other side

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#10900]
```py
a = 1
a = 0 / a  # PLR6104: Use `/=` to perform an augmented assignment directly
```
[playground](https://play.ruff.rs/9874300d-a8c6-4e83-a3b9-841935687a53)

in this example, the quickfix converts it to `a /= 0` which causes a runtime `ZeroDivisionError`

**Comment by MichaReiser** [comment]
CC: @ lshi18

**Comment by lshi18** [comment]
Hi @MichaReiser, I think that [this implementation](https://github.com/astral-sh/ruff/pull/9932/files#diff-9627d6862661b7c0b6d946bfeb654fdf89bbe3f78d272be2c4df23964da961c7R107) is only correct for commutative binary operators, but fails for others such as "/". 

And thank @charliermarsh for completing the work in #9932 .

**Comment by NeilGirdhar** [comment]
@charliermarsh Is this fixed for:
```python
test_string = "desired = " + test_string  # False positive PLR6104
```

**Comment by charliermarsh** [comment]
No, I'll fix that separately.
